From 6e766c6ac5dc4b991b83d9876ad58bde474ab754 Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Wed, 9 Jan 2019 14:39:12 +0900
Subject: [PATCH 1/8] docs: add desc of overview of SPP secondary

Add descriptions of overview of SPP secondary processes in design
section.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 docs/guides/design/spp_secondary.rst | 28 +++++++++++++++++++++++++---
 1 file changed, 25 insertions(+), 3 deletions(-)

diff --git a/docs/guides/design/spp_secondary.rst b/docs/guides/design/spp_secondary.rst
index d83ac6a..b3bb270 100644
--- a/docs/guides/design/spp_secondary.rst
+++ b/docs/guides/design/spp_secondary.rst
@@ -2,15 +2,37 @@
     Copyright(c) 2019 Nippon Telegraph and Telephone Corporation
 
 
-.. _spp_overview_design_spp_secondary:
+.. _spp_design_spp_secondary:
 
 SPP Secondary
 =============
 
-SPP secondary process is ...
+SPP secondary process is a worker process in client-server multp-process
+application model. Basically, the role of secondary process is to connenct
+each of application running on host, containers or VMs for packet forwarding.
+Spp secondary process forwards packets from source port to destination port
+with DPDK's high-performance forwarding mechanizm. In other word, it behaves
+as a cable to connect two patches ports.
 
+All of secondary processes are able to attach ring PMD and vhost PMD ports
+for sending or receiving packets with other processes. Ring port is used to
+communicate with a process running on host or container if it is implemented
+as secondary process to access shared ring memory.
+Vhost port is used for a process on container or VM and implemented as primary
+process, and no need to access shared memory of SPP primary.
 
-.. _spp_overview_design_spp_secondary_nfv:
+In addition to the basic forwarding, SPP secondary process provides several
+networking features. One of the typical example is packet cauture.
+``spp_nfv`` is the simplest SPP secondary and used to connect two of processes
+or other feature ports including PCAP PMD port. PCAP PMD is to dump packets to
+a file or retrieve from.
+
+There are more specific or funcional features than ``spp_nfv``. ``spp_vf`` is
+a simple pseudo SR-IOV feature for classifying or merging packets.
+``spp_mirror`` is to duplicate incoming packets to several destination ports.
+
+
+.. _spp_design_spp_secondary_nfv:
 
 spp_nfv
 -------
-- 
2.7.4

